app/javascript/elm/src/Main.elm

```diff
@@ -1145,16 +1155,26 @@ viewSensorFilter sensors selectedSensorId tooltipIcon =
             , placeholder "sensor"
             , type_ "text"
             , name "sensor"
-            , Popup.clickWithoutDefault (ShowPopup (Sensor.labelsForParameter sensors selectedSensorId) "sensors" (Sensor.sensorLabelForId sensors selectedSensorId))
+            , Popup.clickWithoutDefault (ShowPopup (Sensor.labelsForParameter sensors selectedSensorId) "sensors" (Sensor.sensorLabelForId sensors selectedSensorId) Popup.SensorPopupShown)
             , value (Sensor.sensorLabelForId sensors selectedSensorId)
             , autocomplete False
             ]
             []
         , label [ for "sensor" ] [ text "sensor:" ]
         , Tooltip.view Tooltip.sensorFilter tooltipIcon
+        , viewSensorPopup isPopupExtended popup popupStatus
         ]
 
 
+viewSensorPopup : Bool -> Popup.Popup -> Popup.PopupStatus -> Html Msg
+viewSensorPopup isPopupExtended popup popupStatus =
+    if Popup.isSensorPopupShown popupStatus then
+        Popup.view TogglePopupState SelectSensorId isPopupExtended popup
+
+    else
+        text ""
```

Comment:
I don't know the contex, so maybe this change is needed for something else, but just looking at this commit it seems like there is no reason to have `viewSensorPopup ` and `viewParameterPopup `. This could be just:
```
viewWhateverPopup : Bool -> Popup.Popup -> Popup.PopupStatus -> Html Msg
viewWhateverPopup isPopupExtended popup popupStatus =
    if Popup.isWhateverShown popupStatus then
        Popup.view TogglePopupState SelectSensorId isPopupExtended popup

     else
        text ""
```

but then it doesn't make sense to have this `Popup.PopupStatus` since `Popup.view` already knows that it should show `text ""` if no popup is selected (see in popup module that `Popup.Popup` has two states and one of them is `None`)

Reply:
the thing is that I need to know not just if there's a popup, but which popup it is. hence all the changes.